packages/cli/src/lib/implementation/utils.ts

```diff
@@ -23,3 +24,20 @@ export function coerceArray<T extends string>(param: T | T[] = []): T[] {
     ...new Set(toArray(param).flatMap((f: T) => f.split(',') as T[]) || []),
   ];
 }
+
+export async function withAsyncConfig<T>(
+  unconfiguredCli: Argv<T>,
+  optionsName: string,
+  options: Options = {},
+  transform?: (c: string) => Promise<T>,
+) {
+  // create a yargs CLI only configured to parse the config file
+  const args = await unconfiguredCli
+    .options({ [optionsName]: options })
+    .parseAsync();
+  const configJson = transform
+    ? ((await transform(args[optionsName] as string)) as object)
+    : {};
+  // configure cli with config
+  return unconfiguredCli.config(configJson);
+}
```

Comment:
The more I see this, the less I like it :/ It feels very hacky and unintuitive, the previous solution where it was all the config middleware's responsibility.